Dockerfile-bcoin

```diff
@@ -1,30 +1,48 @@
 FROM nfnty/arch-mini:latest AS base
 
-RUN mkdir -p /code/node_modules/bcoin /data
-WORKDIR /code/node_modules/bcoin
-ENTRYPOINT ["node"]
-CMD ["/code/scripts/docker-bcoin-init.js"]
-
 RUN pacman -Sy --noconfirm archlinux-keyring && \
     pacman -Syu --noconfirm nodejs npm && \
     rm /var/cache/pacman/pkg/*
 
 FROM base AS build
 # Install build dependencies
-RUN pacman -Syu --noconfirm base-devel unrar git python2
-#HACK: Node-gyp needs python
-RUN ln -s /usr/bin/python2 /usr/bin/python
+# Note: node-gyp needs python
+RUN pacman -Syu --noconfirm base-devel unrar git python2 \
+    && ln -s /usr/bin/python2 /usr/bin/python
 
-ARG repo=https://github.com/bcoin-org/bcoin.git
 ARG branch=master
-#HACK: Dingle this to trigger rebuilds
+
+# use this to bust the build cache
 ARG rebuild=0
 
 # Install bcoin
-RUN git clone --branch $branch $repo /code/node_modules/bcoin
-RUN npm install --production
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/bcoin.git /code/node_modules/bcoin \
+    && pushd /code/node_modules/bcoin \
+    && npm install --production \
+    && popd
+
+# Install bmultisig
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/bmultisig.git /code/node_modules/bmultisig \
+    && pushd /code/node_modules/bmultisig \
+    && npm install --production \
+    && popd
+
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/blgr /code/node_modules/blgr \
+    && pushd /code/node_modules/blgr \
+    && npm install --production \
+    && popd
 
 #TODO: Inherit from official image
 FROM base
+
+RUN mkdir -p /code/node_modules/bcoin /data
+
 COPY --from=build /code/node_modules/bcoin /code/node_modules/bcoin
+COPY --from=build /code/node_modules/bmultisig /code/node_modules/bmultisig
+COPY --from=build /code/node_modules/blgr /code/node_modules/blgr
 COPY ./scripts/ /code/scripts
+
+CMD ["/code/scripts/entrypoint.sh"]
```

Comment:
@Thann let us know what you think about moving `CMD` back to the bottom and using the entrypoint shell. This should be nice as it gives us more flexibility for the setup, though maybe this is something best done in docker-compose if we don't want to be married to that setup in the image?